go/tendermint/abci/mux.go

```diff
@@ -481,6 +555,13 @@ func (mux *abciMux) executeTx(ctx *Context, tx []byte) error {
 func (mux *abciMux) CheckTx(req types.RequestCheckTx) types.ResponseCheckTx {
 	ctx := NewContext(ContextCheckTx, mux.currentTime, mux.state)
 
+	if mux.state.haltMode {
```

Comment:
This could be done in `executeTx` as that is called for both `CheckTx` and `DeliverTx`.

Reply:
:+1: